### PR TITLE
Added Shop by Category & brand

### PR DESCRIPTION
**Pull request by gopalshimpi**

### CODE REVIEW

File: src/app/layout/header/components/brand-menu-dropdown/brand-menu-dropdown.component.html [hunk 1]
```diff
@@ -0,0 +1,14 @@
+<div class="btn-group" dropdown [autoClose]="false" container="body">
+  <button id="button-nested" dropdownToggle type="button" class="btn bg-white dropdown-toggle" aria-controls="dropdown-nested">
+    Shop By
+    <span *ngIf="taxonomies.length!==0">{{taxonomies[0].name}}</span>
```
Comment (pkrawat1) [inline src/app/layout/header/components/brand-menu-dropdown/brand-menu-dropdown.component.html]: What if api only sends the array of size 1 only. Then will it not break at this point.

Can you please add an `*ngIf="taxonomies[0]; let brands_taxonomy"`

File: src/app/layout/header/components/categories-menu-dropdown/categories-components/categories-details/categories-details.component.ts [hunk 2]
```diff
@@ -0,0 +1,18 @@
+import { OnChanges } from '@angular/core/src/metadata/lifecycle_hooks';
+import { Component, OnInit, Input } from '@angular/core';
+
+@Component({
+  selector: 'app-categories-details',
+  templateUrl: './categories-details.component.html',
+  styleUrls: ['./categories-details.component.scss']
+})
+export class CategoriesDetailsComponent implements OnInit, OnChanges {
+@Input() taxons;
```
Comment (pkrawat1) [inline src/app/layout/header/components/categories-menu-dropdown/categories-components/categories-details/categories-details.component.ts]: indentation of `@Input`

File: src/app/layout/header/components/categories-menu-dropdown/categories-components/categories-details/categories-list/categories-list.component.ts [hunk 3]
```diff
@@ -0,0 +1,14 @@
+import { Component, OnInit, Input } from '@angular/core';
+
+@Component({
+  selector: 'app-categories-list',
+  templateUrl: './categories-list.component.html',
+  styleUrls: ['./categories-list.component.scss']
+})
+export class CategoriesListComponent implements OnInit {
+@Input () taxons;
```
Comment (pkrawat1) [inline src/app/layout/header/components/categories-menu-dropdown/categories-components/categories-details/categories-list/categories-list.component.ts]: indentation ?

File: src/app/layout/header/components/categories-menu-dropdown/categories-menu-dropdown.component.html [hunk 4]
```diff
@@ -0,0 +1,15 @@
+<div class="btn-group" dropdown [autoClose]="false" container="body">
+  <button id="button-nested" dropdownToggle categories="button" class="btn bg-white dropdown-toggle" aria-controls="dropdown-nested"
+    data-hover="dropdown">
+    Shop By
+    <span *ngIf="taxonomies.length!==0">{{taxonomies[0].name}}</span>
```
Comment (pkrawat1) [inline src/app/layout/header/components/categories-menu-dropdown/categories-menu-dropdown.component.html]: put `taxonomies[0]` in a template variable

File: src/app/layout/header/components/categories-menu-dropdown/categories-menu-dropdown.component.ts [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+
+  }
+
+  ngOnInit() {
+
+
+  }
+
+  ngOnChanges() {
+    console.log('categeory', this.taxonomies)
```
Comment (pkrawat1) [inline src/app/layout/header/components/categories-menu-dropdown/categories-menu-dropdown.component.ts]: remove all console.logs if Pr is ready

File: src/app/layout/header/header.component.html [hunk 6]
```diff
@@ -18,4 +18,14 @@
       <app-header-cart class="pl-2 border-left" [totalCartItems]="totalCartItems | async"></app-header-cart>
     </div>
   </div>
+
 </nav>
+<nav class="navbar navbar-expand-lg">
```
Comment (pkrawat1) [inline src/app/layout/header/header.component.html]: How about we move this to another component ?

File: src/app/layout/header/components/categories-menu-dropdown/categories-menu-dropdown.component.ts (hunk 5 above)
Comment (gopalshimpi) [reply src/app/layout/header/components/categories-menu-dropdown/categories-menu-dropdown.component.ts]: OK

**Pull request merged**